sdk/go/pulumi/stack_reference.go

```diff
@@ -13,14 +15,26 @@ type StackReference struct {
 	Name StringOutput `pulumi:"name"`
 	// Outputs resolves with exports from the named stack
 	Outputs MapOutput `pulumi:"outputs"`
+
+	// Engine context in which resolve resource references in outputs.
+	Context *Context
```

Comment:
I was reluctant to add this because it's adding a field to an exported type in the SDK, but I don't expect people are constructing this except via NewStackReference.

`unmarshalPropertyValue` requires a valid Pulumi engine Context to handle resource references. Can stacks export a resource as an output, and that would _work_? I don't expect it would.

Reply:
Is there a reason we couldn't make `Context` private. I don't think it's breaking to add a private field to an exported type.

Reply:
I don't see a reason why not. I think it's a breaking change only in the sense that the type is non-constructible by third party packages, but I think that was already the case with the embedded type.

Reply:
I might be wrong, but I believe that all types are constructible. You just might not be able to construct a valid type on your own. For example, the following is valid:
```go
var f os.File
f.Chdir()
````
This is true even though the std definition of `os.File` is 
```go
// File represents an open file descriptor.
type File struct {
	*file // os specific
}
```

I'm pretty sure that given package foo
```go
// Pkg foo
type Foo struct {
Public string
private string
}
```
You can still do ` foo.Foo { Public: "foo" }` in another package. You just can't name `foo.Foo.private`.

Reply:
Some day I will internalize Go's fascination with meaningful-zero-values-by-default.